api/src/main/java/org/eclipse/microprofile/metrics/MetricRegistry.java

```diff
@@ -193,7 +187,24 @@ public static String name(Class<?> klass, String... names) {
      *
      * @since 2.0
      */
-    public abstract Counter counter(String name, Tag... tags);
+    default Counter counter(String name, Tag... tags) {
+        return counter(new MetricID(name, tags));
+    }
+
+    /**
+     * Return the {@link Counter} registered under the {@link MetricID};
+     * or create and register a new {@link Counter} if none is registered.
+     *
+     * If a {@link Counter} was created, a {@link Metadata} object will be registered with the name
+     * and type. If a {@link Metadata} object is already registered with this metric name then that
+     * {@link Metadata} will be used.
+     *
+     * @param metricID the ID of the metric
+     * @return a new or pre-existing {@link Counter}
+     * 
+     * @since 2.4
```

Comment:
Yesterday we basically agreed that we will go with a breaking change release, so all these `@since 2.4` can be updated to 3.0